src/Lean/Elab/Tactic/BVDecide/Frontend/Normalize/Structures.lean

```diff
@@ -54,14 +88,39 @@ where
           trace[Meta.Tactic.bv] m!"Using injEq lemma: {lemmaName}"
           let statement ← mkConstWithLevelParams lemmaName
           relevantLemmas ← relevantLemmas.addTheorem (.decl lemmaName) statement
+        let fields := (getStructureInfo env const).fieldNames.size
+        let numParams := constInfo.numParams
+        for proj in [0:fields] do
+          -- We use the simprocs with pre such that we push in projections eagerly in order to
+          -- potentially not have to simplify complex structure expressions that we only project one
+          -- element out of.
+          let path := mkDiscrKeyFor const numParams proj ``ite 5
+          simprocs := simprocs.addCore path ``applyIteSimproc false (.inl applyIteSimproc)
+          let path := mkDiscrKeyFor const numParams proj ``cond 4
+          simprocs := simprocs.addCore path ``applyCondSimproc false (.inl applyCondSimproc)
       let cfg ← PreProcessM.getConfig
       let simpCtx ← Simp.mkContext
         (config := { failIfUnchanged := false, maxSteps := cfg.maxSteps })
         (simpTheorems := relevantLemmas)
         (congrTheorems := ← getSimpCongrTheorems)
-      let ⟨result?, _⟩ ← simpGoal goal (ctx := simpCtx) (fvarIdsToSimp := ← getPropHyps)
+      let ⟨result?, _⟩ ←
+        simpGoal
+          goal
+          (ctx := simpCtx)
+          (simprocs := #[simprocs])
+          (fvarIdsToSimp := ← getPropHyps)
       let some (_, newGoal) := result? | return none
       return newGoal
 
+  mkDiscrKeyFor (struct : Name) (structParams : Nat) (projIdx : Nat) (controlFlow : Name)
```

Comment:
Maybe leave a comment showing a typical path that’s being created here? (It’s not easy to deduce that the `struct` entry below ends up being the motive of the `ite`)